tests/test_api.py

```diff
@@ -446,3 +447,48 @@ def test_pop():
         respx.get("https://foo.bar/", alias="foobar")
         request_pattern = respx.pop("foobar")
         assert request_pattern.url == "https://foo.bar/"
+
+
+@respx.mock
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "url,params,call_url,call_params",
+    [
+        (
+            "https://foo/",
+            "foo=bar&foo=foo",
+            "https://foo/",
+            [("foo", "bar"), ("foo", "foo")],
+        ),
+        (
+            "https://foo/",
+            b"foo=bar&foo=foo",
+            "https://foo/",
+            [("foo", "bar"), ("foo", "foo")],
+        ),
+        (
+            "https://foo/",
+            [("foo", "bar"), ("foo", "foo")],
+            "https://foo/?foo=bar&foo=foo",
+            None,
+        ),
+        ("https://foo/", {"foo": "bar"}, "https://foo/?foo=bar", None),
+        ("https://foo/", {"foo": "bar"}, "https://foo/", {"foo": "bar"}),
+        (
+            "https://foo/?baz=buz",
+            {"foo": "bar"},
```

Comment:
This works, but not the other way around. I.e. Mocking `https://foo/?foo=bar` with `params={"baz": "buz"}` and calling it with `https://foo/?baz=buz&foo=bar` or `https://foo/?baz=buz` with `params={"foo": "bar"}` is a mismatch.

Reply:
Is this still valid @jocke-l?